services/gitdiff/gitdiff.go

```diff
@@ -182,6 +182,7 @@ var (
 	codeTagSuffix     = []byte(`</span>`)
 )
 var trailingSpanRegex = regexp.MustCompile(`<span\s*[[:alpha:]="]*?[>]?$`)
+var entityRegex = regexp.MustCompile(`&[#]*?[0-9a-z]*$`)
```

Comment:
isn't this missing a terminal `;`?


`&([a-zA-Z0-9]+|#[0-9]{1,6}|#[xX][0-9a-fA-Z]{1,6});` is the regexp to detect entities

Reply:
no because if it has the ; then it hasn't been split up -- this is only to detect ones that are incomplete

Reply:
Er

Reply:
Updated regex to look for uppercase letters too since I see some entities can have those, still less strict because it should match partial ones rather than complete ones -- since HTML is already escaped by definition anything that starts with ```&``` and doesn't end with a ```;``` should be a broken entity

Also added a detailed comment/example as requested which hopefully makes it more clear what it does (it is simple, maybe just not very obvious) but happy to change or update anything to make it more clear for anybody else